### ISSUE TITLE
Existence of src/index.ts file causes ng test failure

### ISSUE THREAD

**Issue by dehru** [angular/angular-cli#3316]
> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
OS X El Capitan

### Versions.
angular-cli: 1.0.0-beta.20-4
node: 5.11.1
os: darwin x64

### Repro steps.
1. ng new something
2. cd something
3. ng test
4. touch src/index.ts
5. ng test

### The log given by the failure.
ERROR in ENOTDIR: not a directory, scandir '/Users/cromed/workspace/indexts/src/index.ts'
29 11 2016 15:25:16.377:INFO [karma]: Karma v1.2.0 server started at http://localhost:9876/
29 11 2016 15:25:16.377:INFO [launcher]: Launching browser Chrome with unlimited concurrency
29 11 2016 15:25:16.436:INFO [launcher]: Starting browser Chrome
29 11 2016 15:25:17.132:INFO [Chrome 54.0.2840 (Mac OS X 10.11.6)]: Connected on socket /#tSz3xWiy9o_NWMKEAAAA with id 59087816
Chrome 54.0.2840 (Mac OS X 10.11.6): Executed 0 of 0 ERROR (0.003 secs / 0 secs)

### Mention any other details that might be useful.
it doesn't seem to matter what is in the file src/index.ts just the existence of that file will cause the above error.

**Comment by filipesilva** [comment]
That's a nasty error. We'll look into it.

**Comment by zackarychapple** [comment]
@dehru the default structure of the cli app does not have an index.ts file in the root of `src`

```
mac-zchapple:~ zchapple$ ng --version
angular-cli: 1.0.0-beta.24
node: 6.3.1
os: darwin x64
```

![image](https://cloud.githubusercontent.com/assets/2133184/21810435/d11b22cc-d719-11e6-86cc-0d92410063d6.png)

**Comment by dehru** [comment]
@zackarychapple thanks for your response.  yes I know by default there is no src/index.ts.  But we are using angular-cli as a base for building some shared components between projects at our company, and we wanted to "export" those components from src/index.ts for import into another angular2 project.  So naturally, we added src/index.ts file and told npm about it in package.json kinda like this: 

src/index.ts
```
export * from './app/header'
export * from './app/navigation'
```

package.json
```
"main": "src/index.ts"
```

And that caused the issue.  The workaround is just don't name that file index.ts ( which is not as intuitive ).  we named it src/mainindex.ts and updated package.json accordingly.  But it is odd that the presence of file src/index.ts would be some kinda nasty secret no-no.
